### ISSUE TITLE
new catalogue data for strongbox

### ISSUE THREAD

**Issue by torkus** [ogri-la/wowman#125]
hey @layday , what fields would you capture in your ideal catalogue? ones that you could reuse with `instawow`? If I can make the catalogues appealing to both of us, chances are another might use them.

`wowman` currently uses: `name`, `label`, `alt-name`, `description`, `source`, `source-id`, `download-count`, `category-list`, `uri` and `updated-date`.

When `wowman` asks for more details from the addon host, it adds `download-uri` and `version` fields.

There is also the optional field `game-track-list` for wowinterface, because that is known at catalogue generation time but not available when asking for details. It's also probably incorrect *(I got a reply from wowi about this but it was a bit of a form response and no reply to a follow up. I can pm you the conversation on reddit if you like)*.

Changes I'm making are:

* removing `alt-name`
* normalising `category-list` into a set of tags shared across addon hosts
* `uri` and `download-uri` will become `url` and `download-url` respectively

**Comment by layday** [comment]
I suppose it helps to think about what functions the catalogue should serve.  In instawow, it's used for searching and reconciliation, and for obtaining source IDs from CurseForge URLs.  The `name` is used for searching (I could also see me using your `download-count` to weigh search results), `compatibility` (equivalent to `game-track-list` but populated for all add-ons) and `folders` (no equivalent) are used for reconciliation, and `slug` is used to parse CurseForge URLs (your `uri` could be used for this purpose instead).  You can find the specification of my catalogue [here](https://github.com/layday/instawow/blob/0fae096/instawow/resolvers.py#L49-L59).  If you're unfamiliar with Python type annotations, `compatibility` is an array containing the string literals 'retail' or 'classic', and `folders` is an array of string arrays.  You can find the generated data [here](https://github.com/layday/instawow-data/blob/data/master-catalogue-v1.json).

**Comment by layday** [comment]
antiwinter's wowa has a catalogue [here](https://github.com/antiwinter/scrap/blob/master/wowa/db-curse.json).  It has a `mode` field which I *think* indicates compatibility, with `1` being retail, `2` retail and classic and `3` just classic.  CurseBreaker has several catalogue-alikes which are compiled down to Python bytecode, e.g. [this one](https://github.com/AcidWeb/CurseBreaker/blob/7a361d825dd29b68c89ef0b1a626d52171245c7d/CB/Core.py#L382).  I would have to 'unpickle' them to find out what's in them exactly but the URLs are fairly self-explanatory.

**Comment by torkus** [comment]
this is some real food for thought, thanks @layday - I didn't realise so many others also had their own catalogues

**Comment by layday** [comment]
What did you decide to do?

**Comment by torkus** [comment]
we're all doing roughly the same thing - and more of us are doing it than I realized - so, there is no need to conform here. I'm cleaning up my own catalogue but it will stay reasonably similar to what I have now. I might split the catalogue generation into a separate project with a json schema validation tool to test compatibility. 

That's a little further down the road. Development slowed in Feb as I switched to another project but I'd like to see strongbox released in March
